Reclaim extent block when removing non-root dir

RoyWFHuang opened the pull request:
When removing a directory, the extent block was not reclaimed properly.

This fix ensures that extent blocks for non-root directories are reclaimed during removal. The root directory is excluded from this process, as it serves as the mount point and cannot be removed.

The reclaim mechanism is designed to run when a directory is actually removed, not just when it becomes empty.

close #65

jserv commented on script/test_large_file.sh:
```diff
@@ -0,0 +1,23 @@
+
+
```
Remove unnecessary blank lines.

jserv commented on script/test_large_file.sh:
```diff
@@ -0,0 +1,23 @@
+    echo "test $count"
+    test "$count" -eq 1600 || echo "Failed, file size not matching"
+}
+
```
Ditto.

RoyWFHuang committed:
Reclaim extent block when removing non-root dir

When removing a directory, the extent block was not reclaimed properly.

This fix ensures that extent blocks for non-root directories are
reclaimed during removal. The root directory is excluded from this
process, as it serves as the mount point and cannot be removed.

The reclaim mechanism is designed to run when a directory is actually
removed, not just when it becomes empty.

close #65
